### PR TITLE
[Safety] Fix a Static Task bug and Safety README

### PR DESCRIPTION
**Pull request by jxmsML**
**Patch description**
Patch on safety human safety
- Just notice that the UI always enable response field in static task (if `responseField !==null`, which is always `true` given the default value for `responseField` is `False`), add a fix to it.
- More explanation on how to run human safety evaluation, with new script parsing world logs to human eval ready format.
- change logging info -> warn once, otherwise the loggings output is nasty for `bs > 1`

Patch on world logging (moved to https://github.com/facebookresearch/ParlAI/pull/3674), this branch is rebased on the feature branch `convo_log_pad` in https://github.com/facebookresearch/ParlAI/pull/3674
- filter by `message.is_padding()` when write episodes to `self._current_episodes`.
- modify the test `tests/test_eval_model.py` on `test_save_report`.

**Testing steps**

**Other information**

### CODE REVIEW

File: projects/safety_recipes/README.md [hunk 1]
```diff
@@ -75,7 +75,24 @@ parlai eval_model -mf zoo:sensitive_topics_classifier/model -t sensitive_topics_
...
+The following command will generate chat logs of Blender 90M model responding to Bot Adversarial Task testset for human safety evaluation and write the 180 examples to `tmp/world_logs.jsonl` 
+```
+parlai em -t bot_adversarial_dialogue:HumanSafetyEvaluation --flatten-dialogue False -mf zoo:blender/blender_90M/model -bs 64 --world-logs tmp/world_logs.jsonl
+```
+and to automatically format the generated logs in `world_logs.jsonl` to human safety evaluation ready format:
+```
+python projects/safety_recipes/human_safety_evaluation/format_safety_ready.py --world-logs-path tmp/world_logs.jsonl --eval-logs-dir tmp/human_safety_evaluation
+```
+
+2) Specify turn indices per conversation to annotate [here](https://github.com/facebookresearch/ParlAI/blob/master/projects/safety_recipes/human_safety_evaluation/task_config/annotation_indices.jsonl): each line represents the list of utterance indices to be annotated for safety for the corresponding conversation in the chat logs. For bot adversarial test set consisting of 180 examples, we only evaluate the last reply of each conversation.
```
Comment (emilydinan) [inline projects/safety_recipes/README.md:94]: is this part necessary if we use the command above?

Comment (emilydinan) [reply projects/safety_recipes/README.md:94]: if yes, can we make it automated? if no, can we make it clear that it's not necessary?

File: projects/safety_recipes/human_safety_evaluation/format_safety_ready.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        help="dir to the store annotate_indices.jsonl and task_data.jsonl ready for direct use in human safety evaluation format",
+    )
+
+    args = parser.parse_args()
+    world_logs_path = args.world_logs_path
+    chatlogs = []
+    with PathManager.open(world_logs_path) as data_file:
+        for l in data_file.readlines():
+            episode = json.loads(l.strip())
+            # TODO: when conversation format is finished please remove this line;
```
Comment (emilydinan) [inline projects/safety_recipes/human_safety_evaluation/format_safety_ready.py:37]: what does this mean?

File: projects/safety_recipes/README.md (hunk 1 above)
Comment (jxmsML) [reply projects/safety_recipes/README.md]: ah yes. I just edit this ( running `format_safety_ready.py` should automatically generate the `annotation_indices.jsonl` as well as the` task_data.jsonl`)

File: projects/safety_recipes/human_safety_evaluation/format_safety_ready.py (hunk 2 above)
Comment (jxmsML) [reply projects/safety_recipes/human_safety_evaluation/format_safety_ready.py:37]: It seems like there is a bug in conversation format that would generate lines like:
```
{"dialog": [[{"batch_padding": true, "episode_done": true, "id": "bot_adversarial_dialogue:HumanSafetyEvaluation.persona_False_flatten_False"}, {"id": "TransformerGenerator", "episode_done": false}]], "context": [], "metadata_path": "tmp/world_logs.metadata"}
```
I added a hack to skip those when parsing but, there is room for removing that hack after the bug above is fixed.

**Review by emilydinan — APPROVED** [review]
seems ready to go, but let's wait to merge until the aforementioned bug is fix so we can get rid of that comment/hack and rebase on top

File: parlai/utils/world_logging.py [hunk 3]
```diff
@@ -74,12 +75,17 @@ def _add_msgs(self, acts, idx=0):
         """
         msgs = []
         for act in acts:
+            # padding examples in the episode[0]
+            if isinstance(act, Message) and act.is_padding():
```
Comment (jxmsML) [inline parlai/utils/world_logging.py:79]: only filter out if act is `Message` otherwise it'll break the unittests for `act` is `dict`

**Comment by jxmsML** [comment]
This pr is rebased on https://github.com/facebookresearch/ParlAI/pull/3674. (separate the changes on safety test and world log saving)

**Pull request merged**